src/syft/lib/torch/module.py

```diff
@@ -396,3 +404,81 @@ def debug_sum_layers(self) -> None:
                     if hasattr(v, "sum"):
                         s = v.sum().item()
                         info(f"  Layer {n} sum({k}): {s}")
+
+
+def object2proto(obj: torch.nn.Module, is_child: bool = False) -> Module_PB:
+    proto = Module_PB()
+    if "torch.nn." in type(obj).__module__:
+        proto.module_type = type(obj).__name__
+    else:
+        proto.module_type = f"_USER_DEFINED_MODULE_{type(obj).__name__}"
+
+    proto.module_repr = obj.extra_repr()
+
+    if not is_child:
+        proto.state_dict.CopyFrom(sy.serialize(SyOrderedDict(obj.state_dict())))
+
+    for n, m in obj.named_children():
+        child_proto = object2proto(m, is_child=True)
+        child_proto.module_name = n
+        proto.children.append(child_proto)
+
+    return proto
+
+
+def proto2object(proto: Module_PB) -> torch.nn.Module:
+    is_userdefined = proto.module_type.startswith("_USER_DEFINED_MODULE_")
+
+    if is_userdefined:
+        obj_type = type(
+            proto.module_type.replace("_USER_DEFINED_MODULE_", ""),
+            (torch.nn.Module,),
+            {},
+        )
+    else:
+        obj_type = getattr(torch.nn, proto.module_type)
+
+    args, kwargs = repr_to_kwargs(repr_str=proto.module_repr)
+    obj = obj_type(*args, **kwargs)
+
+    for child_proto in proto.children:
+        obj.add_module(child_proto.module_name, sy.deserialize(child_proto))
+
+    if proto.state_dict.ByteSize() > 0:
+        obj.load_state_dict(sy.deserialize(proto.state_dict))
+
+    return obj
+
+
+GenerateWrapper(
+    wrapped_type=torch.nn.Module,
+    import_path="torch.nn.Module",
+    protobuf_scheme=Module_PB,
+    type_object2proto=object2proto,
+    type_proto2object=proto2object,
+)
+
+
+class ModelExecutor:
```

Comment:
Would you mind adding:
* a test with a very dummy model (in can have like an internal FC of [32, 2]) with random data/labels to add a test that the forward pass is not breaking?
* add a few lines of docstrings?

Reply:
sure, will do.

Reply:
sure, will do.